### PR TITLE
feat(hcl2cdk): support count using TerraformCount

### LINKED ISSUE #2764: Count should be handled with API in convert

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
### Related issue

Fixes #2764

### Description

count attributes used overrides in the past, this PR let's them use the new TerraformCount construct

### Checklist

- [X] I have updated the PR title to match [CDKTF's style guide](https://github.com/hashicorp/terraform-cdk/blob/main/CONTRIBUTING.md#pull-requests-1)
- [X] I have run the linter on my code locally
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation if applicable
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works if applicable
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by team-tf-cdk** [comment]
This PR exceeds the recommended size of 1000 lines. Please make sure you are NOT addressing multiple issues with one PR. Note this PR might be rejected due to its size.

File: packages/@cdktf/hcl2cdk/lib/generation.ts [hunk 1]
```diff
@@ -385,6 +385,34 @@ export async function resource(
     mappedConfig.forEach = t.identifier(forEachIteratorName);
   }
 
+  let countIteratorName: string | undefined;
+  if (count) {
+    countIteratorName = variableName(scope, resource, `${key}_count_iterator`);
```
Comment (ansgarm) [inline packages/@cdktf/hcl2cdk/lib/generation.ts:390]: Is there a reason this is called count "iterator"? The underlying class is also just named `TerraformCount` and does not share anything with the TerraformIterators. Could we just get rid of `_iterator`?

**Review by ansgarm — APPROVED** [review]

File: packages/@cdktf/hcl2cdk/test/__snapshots__/iteration.test.ts.snap [hunk 2]
```diff
@@ -121,20 +121,26 @@ class MyConvertedCode extends cdktf.TerraformStack {
...
+    you need to keep this like it is.*/
+    const awsIamUserLbCountIterator = cdktf.TerraformCount.of(
+      cdktf.Fn.lengthOf(users.value)
+    );
     const awsIamUserLb = new aws.iamUser.IamUser(this, "lb", {
       name: cdktf.Token.asString(
-        cdktf.Fn.element(users.value, cdktf.Token.asNumber("\${count.index}"))
+        cdktf.Fn.element(
+          users.value,
+          cdktf.Token.asNumber(awsIamUserLbCountIterator.index)
```
[the remark this answers is not in the document]
Comment (DanielMSchmidt) [reply packages/@cdktf/hcl2cdk/test/__snapshots__/iteration.test.ts.snap:135]: This is again the data race condition we discussed in different PRs. We are going to tackle it as part of a mob programming once we finalize the current work.

File: packages/@cdktf/hcl2cdk/lib/generation.ts (hunk 1 above)
Comment (DanielMSchmidt) [reply packages/@cdktf/hcl2cdk/lib/generation.ts:390]: Good point, I'll rename it

**Pull request merged**